web/controllers/v1/relay_controller.ex

```diff
@@ -46,6 +46,10 @@ defmodule Cog.V1.RelayController do
     changeset = Relay.changeset(relay, relay_params)
     case Repo.update(changeset) do
       {:ok, relay} ->
+        # If the enabled flag has changed we need to enable/disable the relay
```

Comment:
I think this functionality needs to move out of the controller and closer to the core. That is, whenever we update a relay from _anywhere_, we should be sure that the state of the system is consistent.

One way to do it is to put the database operations (like `Repo.update`, here) behind another Cog-specific interface, which acts as a DDD-style "repository". The only way to update a relay would be to go through that interface, and behind it, we're free to do whatever we want. In this case, we can just make the call to update the status, but in the future it opens up more possibilities (e.g., emitting events that arbitrary system components can listen for and then take action on, etc.).

I started this with triggers  - https://github.com/operable/cog/blob/master/lib/cog/repository/triggers.ex - thinking that each "thing" could perhaps have it's own "repository". That could ultimately change, but it seemed like a decent way to decompose things to start with.